src/Microsoft.OData.Core/UriParser/Parsers/FunctionParameterParser.cs

```diff
@@ -50,6 +50,82 @@ internal static bool TrySplitOperationParameters(string parenthesisExpression, O
             return ret;
         }
 
+        /// <summary>
+        /// Splits the parenthesis expression into two parts (if apply)
+        /// One is the function parameter, the other is key in parenthesis (if exists)
+        /// Be noted, the input expression doesn't contain the beginning "(" and the ending ")"
+        /// </summary>
+        /// <param name="parenthesisExpression">the input expression</param>
+        /// <param name="parameters">the output for parameter part</param>
+        /// <param name="parenthesisKey">the output for key in parenthesis part</param>
+        internal static void SplitOperationParametersAndParenthesisKey(string parenthesisExpression, out string parameters, out string parenthesisKey)
+        {
+            // Be noted, the input expression doesn't contain the first '(' and last ')'.

```

Comment:
Should we have a sanity check to confirm that the input expression does not containe the beginning "(" and the ending ")"?